internal/provider/data_source_groups_test.go

```diff
@@ -0,0 +1,80 @@
+package provider
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccDataSourceGroups(t *testing.T) {
+	email := randomEmail()
+	name := randomName()
+
+	dataSourceName := fmt.Sprintf("data.infra_groups.%s", t.Name())
+
+	resource.UnitTest(t, resource.TestCase{
+		PreCheck:          testAccPreCheck(t),
+		ProviderFactories: testAccProviders(t),
+		Steps: []resource.TestStep{
+			{
+				Config: composeTestConfigFunc(
```

Comment:
We want to preload the test environment with some resources so the data sources can find them in later steps. It's necessary to keep the resources around in each step so Terraform doesn't destroy them